src/pushin.js

```diff
@@ -1,10 +1,90 @@
+const { hasOwnProperty } = Object.prototype;
+
+/**
+ * Set the z-index of each layer so they overlap correctly.
+ *
+ * @param {object} layer
+ * @param {int} total
+ */
+function setZIndex(layer, total) {
+  layer.elem.style.zIndex = total - layer.index;
+}
+
+/**
+ * Set element scale.
+ *
+ * @param {HtmlElement} elem
+ * @param {Number} value
+ */
+function setScale(elem, value) {
+  const { style } = elem;
+  const scaleString = `scale(${value})`;
+  style.webkitTransform = scaleString;
+  style.mozTransform = scaleString;
+  style.msTransform = scaleString;
+  style.oTransform = scaleString;
+  style.transform = scaleString;
+}
+
+/**
+ * Get the initial scale of the element at time of DOM load.
+ *
+ * @param {Element} elem
+ * @return {Number} scaleX
+ */
+function getElementScaleX(elem) {
+  const transform = window.getComputedStyle(elem).getPropertyValue('transform');
+
+  let scaleX = 1;
+  if (transform && transform !== 'none') {
+    const match = transform.match(/[matrix|scale]\(([\d,.\s]+)/);
+    if (match && match[1]) {
+      const matrix = match[1].split(', ');
+      scaleX = parseFloat(matrix[0]);
+    }
+  }
+
+  return scaleX;
+}
+
+function getSpeed(elem) {
+  const speed = hasOwnProperty.call(elem.dataset, 'pushinSpeed')
+    ? elem.dataset.pushinSpeed
+    : null;
+  return speed || 8;
+}
+
+function showDebugger() {
+  const scrollCounter = document.createElement('div');
+  scrollCounter.classList.add('pushin-debug');
+
+  const scrollTitle = document.createElement('p');
+  scrollTitle.innerText = 'Pushin.js Debugger';
+  scrollTitle.classList.add('pushin-debug__title');
+
+  const debuggerContent = document.createElement('div');
+  debuggerContent.classList.add('pushin-debug__content');
+  debuggerContent.innerText = `Scroll position: ${window.pageYOffset}px`;
+
+  scrollCounter.appendChild(scrollTitle);
+  scrollCounter.appendChild(debuggerContent);
+
+  document.body.appendChild(scrollCounter);
+
+  window.addEventListener('scroll', () => {
+    debuggerContent.innerText = `Scroll position: ${Math.round(
+      window.pageYOffset
+    )}px`;
+  });
+}
+
```

Comment:
@arturovt it looks like a lot of extra code is being introduced here. These functions should be part of the PushIn class. Any idea what happened here?

Reply:
Reverted it back, the lint was telling that methods should be functions because they're not using `this`.

Reply:
Oh interesting, that is a good point.
Thanks for reverting it back for now. I might look into how to refactor that at some point.